src/actions/filterActions.js

```diff
@@ -44,12 +60,17 @@ export const removeFilter = (history, relativeFilterPath) => (oldFilter, filterT
 }
 
 const applyFilter = (history) => (oldFilter, newFilter) => (dispatch) => {
-  if (!newFilter.incidentId) {
-    throw new Error('Need to filter on incidentId!')
-  }
-  if (!deepEquals(oldFilter, newFilter)) {
-    dispatch(changeEventFilter(history)(newFilter))
-    dispatch(eventActions.fetchEvents(newFilter))
+  if (newFilter.incidentId) {
+    if (!deepEquals(oldFilter, newFilter)) {
+      dispatch(changeEventFilter(history)(newFilter))
+      dispatch(eventActions.fetchEvents(newFilter))
+    }
+  } else {
+    const endTime = newFilter.endTime ? DateTime.fromJSDate(new Date(newFilter.endTime)) : DateTime.local()
+    const startTime = newFilter.startTime ? DateTime.fromJSDate(new Date(newFilter.startTime)) : DateTime.local().minus({days: 1})
```

Comment:
It may be better not to tie the hardcoded start and end times in a filter to the lack of incidentId on the filter; moving these calculations to the componentDidMount method of uncorrelatedEvents.js and including them in the filter passed into the dispatch from that method would give the same effect, but it would put the code for determining the date range shown by default on the uncorrelatedEvents component in the same file as that component, which seems preferable.